lib/formats/odata.js

```diff
@@ -418,6 +430,48 @@ const singleRowToOData = (fields, row, domain, originalUrl, query) => {
   });
 };
 
+const systemFields = new Set([
+  '__system/submissiondate',
+  '__system/updatedat',
+  '__system/submitterid',
+  '__system/submittername',
+  '__system/attachmentspresent',
+  '__system/attachmentsexpected',
+  '__system/status',
+  '__system/reviewstate',
+  '__system/deviceid',
+  '__system/edits',
+  '__system/formversion',
+]);
+
+// Validates $select query parameter including metadata properties and returns list of FormFields
+const filterFields = (formFields, select, table) => {
+  const filteredFields = new Set();
+  const fieldMap = formFields.reduce((map, field) => ({ ...map, [field.path.toLowerCase()]: field }), {});
+  for (const property of select.split(',').map(p => p.trim().toLowerCase())) {
+    // validate metadata properties. __system/.. properties are only valid for Submission table
+    if (property.startsWith('__')) {
+      if (!(property === '__id' || (table === 'Submissions' && systemFields.has(property))))
+        throw Problem.user.propertyNotFound({ field: property });
+    } else {
+      // we have to include parents fields in the result to handle grouped fields
+      const propSegments = property.split('/');
+      let path = '';
+      for (let i = 0; i < propSegments.length; i+=1) {
+        path += `/${propSegments[i]}`;
+        const field = fieldMap[path];
+        if (!field) throw Problem.user.propertyNotFound({ field: property });
+        // it's ok to include field with repeat type so that user can have navigation link
+        // but child field of a repeat field is not supported
+        if (field.type === 'repeat' && i < propSegments.length - 1) throw Problem.user.unsupportedODataSelectField({ field: property });
```

Comment:
Yes, you are right if $expand is used with $select then empty repeat groups are returned. 
Since only `$expand=*` is allowed, I can throw unsupported feature problem if user provides both `$select` and `$expand` or we can simply ignore `$select` option if `$expand` is also present, what do you recommend?

Reply:
Added validation in `endpoint.js`